docusaurus/docs/iOS/uikit/components/voice-recording.md

```diff
@@ -0,0 +1,148 @@
+
+:::note
+Voice Recordings are available since version 4.32.0.
+:::
+
+VoiceRecording creation and playback is by default disabled. If you would like to enable it, you can do so by setting the Components `voiceRecordingEnabled` property to `true` like below:
+```swift
+Components.default.voiceRecordingEnabled = true
+```
+
+:::note
+When the VoiceRecording feature is disabled, any message containing VoiceRecordings will render them as regular audio attachments.
+:::
+
+## UI Flows
+### Recording
+The recording flow is being presented on the `ComposerContentView` of the `ComposerVC` and it's being managed by `VoiceRecordingVC`. The `VoiceRecordingVC` takes care of the following:
+- Update the UI based on the recordings state (for example recording, locked or in preview)
+- Update the UI with recording related information (for example duration, current playback time)
+- Coordinate information flow from audio recording and player.
+- Communicate with `ComposerVC` in order to add a VoiceRecording as attachment or send it.
+
+The main UI components that `VoiceRecordingVC` manages are the following:
+#### recordingTipView
+A view that is being used to display a tip to user when the tap duration on the recording button wasn't long enough (the display of this view relates to `RecordButton.minimumPressDuration`).
+
+#### slideToCancelView
+A view that during an unlocked recording, displays information on how to cancel the active recording.
+
+#### recordingIndicatorView
+A view that indicates to the user that we are currently recording audio.
+
+#### lockIndicatorView
+A view that indicates to the user if the currently active recording is locked or not.
+
+#### liveRecordingView
+A view that during a locked recording, displays information about the active recording or its preview.
+
+#### bidirectionalPanGestureRecogniser
+The gestureRecogniser used to identify touch movements in the horizontal or vertical axis.
+ 
+### Presentation
+VoiceRecordings are being presented using the `voiceRecordingAttachmentView` that is defined and configurable in Components:
+```swift
+Components.default.voiceRecordingAttachmentView = ChatMessageVoiceRecordingAttachmentListView.ItemView.self
+```
+
+The view manages the visibility and state of all UI components (e.g play/pause button, waveform visualization, playbackRate button and name label).
+
+For anything else (interaction and event handling) it relies on its `presenter` property.
+
+The `presenter` is of type `ChatMessageVoiceRecordingAttachmentListView.ItemViewPresenter` and it's responsible:
+1. To handle user interactions on the `voiceRecordingAttachmentView`
+2. To handle events/updates that are being sent by the AudioPlayer during the voiceRecording's playback.
```

Comment:
I would actually omit this information for now since it is implementation details, but let me know WDYT 👍 

```suggestion
```

Reply:
I'm happy to remove it. The reason for adding it here is to provide more information and undestanding how it works. Given though that the presenter is internal that's not necessary